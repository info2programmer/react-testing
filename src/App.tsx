


import { useState } from 'react'
import './App.css'

function App() {
  const [data, setdata] = useState<string>("")

  return (

    <>
      <h1>Jest Testing </h1>
      <button onClick={() => setdata("Hello")}>Update</button>
      <p>{data}</p>
    </>
  )
}

export default App
