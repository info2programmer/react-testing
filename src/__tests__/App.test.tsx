import { expect, test } from "vitest";
import { render } from '@testing-library/react';
import App from "../App";


test("snapshot for app component", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
})