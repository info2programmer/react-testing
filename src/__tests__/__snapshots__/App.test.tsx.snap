// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot for app component 1`] = `
<div>
  <h1>
    Jest Testing 
  </h1>
  <button>
    Update
  </button>
  <p />
</div>
`;
